arch-design/multicloud/BucketManagement.md

```diff
@@ -0,0 +1,369 @@
+# Bucket management in soda-multicloud
+
+
+**Author(s)**: [Pravin Ranjan](https://github.com/PravinRanjan10), [Vineela Pachchipulusu]()
+
+
+## Motivation and background
+
+In the cloud world, every cloud vendor provides storage space to users. Generally in terms of bucket and folder. A user can store the data/objects in a specific bucket or folder.
+
+SODA multi-cloud(gelato) is a project which manages multiple cloud vendors from one place. And also interfaces to use cloud services. So it is important for soda to provide bucket management support for various cloud vendors. So that user can create and manage buckets from one place(soda multi-cloud).
+
+## Goals
+The document provides the use case, design and implementation guidelines for supporting bucket management from SODA platform. It includes creation, deletion, object uploading, migration etc. in various cloud providers.
+
+### Non-Goals
+NA
+
+### Requirement Analysis
+
+##### Functional Requirements
+
+1. Create buckets in various cloud providers. Ex: Aws, gcp, azure
+
+2. Delete bucket from various cloud providers. Ex: Aws, gcp, azure
+
+3. Upload objects in bucket
+
+4. Set lifecycles rules on objects
+
+5. Migrate objects from one location to different locations
```

Comment:
can it discover existing (created in the cloud) buckets and manage them there-on ?

Reply:
Discovering and syncing points are considered. But its currently out of scope for this PR/design. There will be separate document for discovering and syncing of created/existing buckets.

Reply:
Currently, it won't discover existing buckets. But we have considered this discovring and syncing for later releases.